core/src/main/scala/kafka/admin/ConfigCommand.scala

```diff
@@ -117,6 +117,10 @@ object ConfigCommand {
             "Invalid entity config: all configs to be added must be in the format \"key=val\".")
     val props = new Properties
     configsToBeAdded.foreach(pair => props.setProperty(pair(0).trim, pair(1).trim))
+    if (props.containsKey(LogConfig.MessageFormatVersionProp)) {
+      println(s"WARNING: The configuration ${LogConfig.MessageFormatVersionProp}=${props.getProperty(LogConfig.MessageFormatVersionProp)} is specified. " +
+        s"This configuration will be ignored if it is not compatible with the inter broker protocol configuration in the Kafka cluster.")
```

Comment:
Hi Jun, this comment seems not accurate based on the current code. What we actually check is the message format version, not Kafka version. For example, the current code will allow the message.format.version to be set to 0.9.0 even if the inter.broker.protocol.version is 0.8.2, because the underlying message.format.versions of those two Kafka versions are the same. At some point I thought there was a use case for this, but I cannot think of any now.

It is probably better to enforce the check as you described. I will make this change in the updated patch.